src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -54,27 +51,25 @@ public int getSize() {
     }
 
     private int getIndex(K key, int tableLength) {
-        int hashCode = (key != null)
-                ? key.hashCode()
-                : 0;
-        int index = Math.abs(hashCode) % tableLength;
-        return index;
+        return key == null ? 0 : Math.abs(key.hashCode()) % tableLength;
     }
 
     private void resize() {
-        int newCapacity = table.length * DEFAULT_RESIZE_COEFFICIENT;
-        Node<K, V>[] newTable = new Node[newCapacity];
-        for (Node<K, V> node : table) {
-            while (node != null) {
-                Node<K, V> next = node.next;
-                int newIndex = getIndex(node.key, newCapacity);
-                node.next = newTable[newIndex];
-                newTable[newIndex] = node;
-                node = next;
+        if ((double) size / table.length > DEFAULT_LOAD_FACTOR) {
+            int newCapacity = table.length * DEFAULT_RESIZE_COEFFICIENT;
+            Node<K, V>[] newTable = new Node[newCapacity];
+            for (Node<K, V> node : table) {
+                while (node != null) {
+                    Node<K, V> next = node.next;
+                    int newIndex = getIndex(node.key, newCapacity);
+                    node.next = newTable[newIndex];
+                    newTable[newIndex] = node;
+                    node = next;
+                }
```

Comment:
You can store previous table in some local variable, make new table as current and just use put method to save nodes in new table